README.md

```diff
@@ -73,7 +73,23 @@ Currently we support two ways to deploy RIG: using Docker and using classical Er
 
 You can find list of all environment variables in [operator guide](guides/operator-guide.md).
 
-### Deployment using Docker
+### Deployment using Docker & Distillery
+
+We are using Distillery releases to be able to run production version of RIG correctly optimized and tuned. Distillery has multiple entrypoints that are used for release:
+
+* `rel/config.exs` - Includes configuration for different types of environment (e.g. dev, prod, staging, etc.). Here Distillery finds which umbrella applications to compile and to use custom `vm.args` config file.
+* `vm.args` - Specifies options for Erlang VM, including name of the node and it's cookie. **Note:** cookie is also set in `rel/config.exs`, however it's just a dummy value and real value for it comes from this file (reason behind is that Distillery needs cookie set in `rel/config.exs` and Erlang in `vm.args` file, skipping either of those leads to error in build/production phase).
+* `config/` - Configuration for RIG itself (most of it can be set by environment variables on runtime, see [operator guide](guides/operator-guide.md)).
+* `apps/` - RIG's source folders.
+
+All these files are compiled together with an external dependencies and released as a tarball.
+
+By running built Docker image in container, release is executed in a foreground.
+
+There are multiple Dockerfiles that can be build:
+
+* `Dockerfile` - Basic version, excludes Java, thus AWS Kinesis is not possible to use.
```

Comment:
I wouldn't specify this here. Actually, we don't know if we'll ever need Java in the main image (I really hope we don't). What about
```
Dockerfile -- Standard image, suitable for on-premise use.
aws.dockerfile -- AWS-flavored version that let's RIG integrate with AWS services like Amazon Kinesis.
```